js/forum/src/components/EmojiAreaButton.js

```diff
@@ -35,9 +30,9 @@ export default class EmojiAreaButton extends Component {
             useInternalCDN: false, // Use the same CDN as Flarum so images are not fetched twice
             buttonTitle: 'Emoji', // The default text includes something with TAB, even for the standalone version where it is useless
             events: { // Listen for clicks to sync with Flarum editor
-                emojibtn_click: function (button, event) {
-                    var shortcode = button.data('name');
-                    editor.insertAtCursor(shortcode);
+                emojibtn_click: button => {
+                    const shortcode = button.data('name');
+                    this.props.textEditor.insertAtCursor(shortcode);
```

Comment:
Then you'd use `this.textEditor` and I'd argue you shouldn't need to define the shortcode variable and you can just put `button.data('name')` in here.